private/tools/java/com/github/bazelbuild/rules_jvm_external/Coordinates.java

```diff
@@ -73,11 +84,11 @@ public String getClassifier() {
   }
 
   public Coordinates setClassifier(String classifier) {
-    return new Coordinates(getGroupId(), getArtifactId(), getExtension(), classifier, getVersion());
+    return new Coordinates(getGroupId(), getArtifactId(), getExtension(), classifier, getVersion(), dirVersion);
```

Comment:
For consistency with the rest of the call, use `getDirVersion()`. Here and below in `setExtension()`.